### PR TITLE
Add a test to check that all exported models have a portable unique name

### PR DESCRIPTION
**Pull request by bshepherdson**
Either an external name (eg. a database path) or an entity_id column.
Unexported models are explicitly listed as such here.

This makes it impossible to forget to address a unique ID for any new
model.

Related to epic #22641

### CODE REVIEW

File: test/metabase/models/entity_id_test.clj [hunk 1]
```diff
@@ -0,0 +1,68 @@
+(ns metabase.models.entity-id-test
```
Comment (qnkhuat) [inline test/metabase/models/entity_id_test.clj:1]: I believe this should be in ee ns?

File: test/metabase/models/entity_id_test.clj [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  "Entities with external names, so they don't need a generated entity_id."
+  #{;; Databases have external names based on their URLs; tables are nested under databases; fields under tables.
+    metabase.models.database.DatabaseInstance
+    metabase.models.table.TableInstance
+    metabase.models.field.FieldInstance
+    ;; Settings have human-selected unique names.
+    metabase.models.setting.SettingInstance
+    ;; Dependencies are serialized but no extra ID is necessary since they're just many-to-many links between entities
+    ;; with unique IDs, and don't need unique IDs of their own.
+    metabase.models.dependency.DependencyInstance})
```
Comment (qnkhuat) [inline test/metabase/models/entity_id_test.clj:22]: nit: this is not sorted

File: test/metabase/models/entity_id_test.clj [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    metabase.models.secret.SecretInstance
+    metabase.models.session.SessionInstance
+    metabase.models.task_history.TaskHistoryInstance
+    metabase.models.timeline_event.TimelineEventInstance
+    metabase.models.user.UserInstance
+    metabase.models.view_log.ViewLogInstance
+    metabase_enterprise.sandbox.models.group_table_access_policy.GroupTableAccessPolicyInstance})
+
+(deftest comprehensive-entity-id-test
+  (is (empty? (->> (extenders IModel)
```
Comment (qnkhuat) [inline test/metabase/models/entity_id_test.clj:63]: nit: I'd do something like this
```clojure
(deftest comprehensive-entity-id-test
  (doseq [model (->> (extenders IModel)
                     (remove entities-not-exported)
                     (remove entities-external-name))]
    (testing (format "Model %s should have entity_id property or be explicitly listed as having an external name, or explicitly listed as excluded from serialization" (.getSimpleName model))
      (is (= true (-> (.newInstance model)
                      toucan.models/properties
                      :entity_id))))))
```
     
having an explicit test name make it easy to debug, for example if a model false we will have something like 

`Model CollectionBookmarkInstance should have entity_id property or be explicitly listed as having an external name, or explicitly listed as excluded from serialization`, 

which I think is easier to understand.

This is more or less a style thing, so up to you

File: test/metabase/models/entity_id_test.clj (hunk 1 above)
Comment (bshepherdson) [reply test/metabase/models/entity_id_test.clj]: It is now.

File: test/metabase/models/entity_id_test.clj (hunk 2 above)
Comment (bshepherdson) [reply test/metabase/models/entity_id_test.clj]: I didn't intend to sort this list, but rather to capture the semantic order databases -> tables -> fields.
Should I just sort it instead?

File: test/metabase/models/entity_id_test.clj (hunk 3 above)
Comment (bshepherdson) [reply test/metabase/models/entity_id_test.clj]: I like that better, yeah!

File: test/metabase/models/entity_id_test.clj (hunk 2 above)
Comment (qnkhuat) [reply test/metabase/models/entity_id_test.clj]: ah, then I think we should leave that unsorted.

**Review by qnkhuat — APPROVED** [review]
Looks good! there is an unsorted namespace tho.

You probably know this but you could run `clj -X:dev:ee:ee-dev:drivers:drivers-dev:test:namespace-checker` to check ns locally

**Pull request merged**